allennlp/training/trainer_pieces.py

```diff
@@ -41,44 +41,76 @@ def from_params(
         recover: bool = False,
         cache_directory: str = None,
         cache_prefix: str = None,
+        model: Model = None,
+        embedding_sources_mapping: Dict[str, str] = None,
+        extend_vocab: bool = False,
     ) -> "TrainerPieces":
         all_datasets = training_util.datasets_from_params(params, cache_directory, cache_prefix)
-        datasets_for_vocab_creation = set(params.pop("datasets_for_vocab_creation", all_datasets))
 
-        for dataset in datasets_for_vocab_creation:
-            if dataset not in all_datasets:
-                raise ConfigurationError(f"invalid 'dataset_for_vocab_creation' {dataset}")
+        vocabulary_params = params.pop("vocabulary", {})
 
-        logger.info(
-            "From dataset instances, %s will be considered for vocabulary creation.",
-            ", ".join(datasets_for_vocab_creation),
-        )
+        if model:
+            if params.pop("model", None):
+                logger.warning(
+                    "You passed parameters for the model in your configuration file, but we "
+                    "are ignoring them, using instead the model parameters in the archive."
+                )
+
+            if vocabulary_params.get("directory_path", None):
+                logger.warning(
+                    "You passed `directory_path` in parameters for the vocabulary in "
+                    "your configuration file, but it will be ignored. "
```

Comment:
Why?  `s/ignored. "/ignored, because we already have a model with a vocabulary."`